### PR TITLE
chore: update Numpy and the matching syntax

### LINKED ISSUE #245: [FEATURE] Update Numpy 1.20 to 1.21 for mypy
We have added types to ndarray, which requires NumPy 1.21 support. We are supposed to make this update for mypy entry in the pre-commit-config.

Probably will hit boost-histogram, too.

### PR DESCRIPTION
**Pull request by amangoel185**
Close #245 when merged.

### CODE REVIEW

File: src/hist/plot.py [hunk 1]
```diff
@@ -608,7 +615,7 @@ def _plot_ratiolike(
         if fit_fmt is not None:
             parnames = list(inspect.signature(other).parameters)[1:]
             popt, pcov = bestfit_result
-            perr = np.sqrt(np.diag(pcov))
+            perr = np.sqrt(np.diag(pcov))  # type: ignore
```
Comment (henryiii) [inline src/hist/plot.py:618]: Can we switch to using `np.diagonal`? `np.diag` is a Matlab compat function, which is why it's untyped, I think.

File: src/hist/plot.py [hunk 2]
```diff
@@ -651,7 +658,9 @@ def _plot_ratiolike(
             )
 
         elif view == "pull":
-            pulls = (hist_values - compare_values) / hist_values_uncert
+            pulls: np.typing.NDArray[Any] = (
```
Comment (henryiii) [inline src/hist/plot.py:661]: We should check and see why this requires a type now - one of the three input values here (hist_values, compare_values, hist_values_uncert) might be missing a type. Don't have to happen in this PR though.

File: src/hist/plot.py (hunk 1 above)
Comment (amangoel185) [reply src/hist/plot.py:618]: sure, let me check.

**Review by henryiii — APPROVED** [review]

File: src/hist/plot.py (hunk 2 above)
Comment (henryiii) [reply src/hist/plot.py:661]: From @amangoel185:

`reveal_type` on the values gives:

```
src/hist/plot.py:661: note: Revealed type is "numpy.ndarray[Any, Any]" # hist_values
src/hist/plot.py:662: note: Revealed type is "numpy.ndarray[Any, numpy.dtype[Any]]" # compare_values
src/hist/plot.py:663: note: Revealed type is "numpy.ndarray[Any, numpy.dtype[Any]]" # hist_values_uncert
```

**Pull request merged**